hr_timesheet_hours_billed/models/hours_billed.py

```diff
@@ -0,0 +1,45 @@
+from odoo import api, fields, models
+
+
+class AccountAnalyticLine(models.Model):
+    _inherit = "account.analytic.line"
+
+    approved = fields.Selection(
+        selection=[("no", "No"), ("yes", "Yes")],
+        default="no",
+        required=True,
+    )
+    approved_uid = fields.Many2one(
+        "res.users",
+        string="Approved by",
+        index=True,
+        tracking=True,
+        compute="_compute_approved_uid",
+        store=True,
+    )
+    approved_date = fields.Date(
+        copy=False, compute="_compute_approved_date", store=True
+    )
+    unit_amount_billed = fields.Float(string="Hours Billed")
+
+    # set approved_uid if approved state change to 'yes'
+    @api.depends("approved")
+    def _compute_approved_uid(self):
+        for record in self:
+            if record.approved == "yes":
+                record.approved_uid = self.env.user
+
+    # set approved_date if approved state change to 'yes'
+    @api.depends("approved")
+    def _compute_approved_date(self):
+        for record in self:
+            if record.approved == "yes":
+                record.approved_date = fields.Datetime.today()
```

Comment:
we can set `approved_date` and `approved_user_id` by inverse method of `approved` field, e.g:

inverse="_inverse_approved"

```python
def _inverse_approved(self):
    now = fields.Datetime.now()
    user_id = self.env.user.id
    for record in self.filtered(lambda rec: rec.approved):
        record.write({"approved_date": now, "approved_user_id": user_id})

Reply:
> we can set `approved_date` and `approved_user_id` by inverse method of `approved` field, e.g:
> 
> inverse="_inverse_approved"
> 
> ```python
> def _inverse_approved(self):
>     now = fields.Datetime.now()
>     user_id = self.env.user.id
>     for record in self.filtered(lambda rec: rec.approved):
>         record.write({"approved_date": now, "approved_user_id": user_id})
> ```

Why do we need an inverse method? We are not updating these field manually

Reply:
@isserver1 How is the field updated?

Reply:
> @isserver1 How is the field updated?

We cannot set date of approval explicitly as well as user who approved the record

Reply:
> Why do we need an inverse method? We are not updating these field manually

it will work anyway.

Reply:
@solo4games please, use inverse method to set  approved_date and approved_user_id, thanks